src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -6290,6 +6315,181 @@ .maxstack  2
 }");
         }
 
+        [Fact]
+        public void IEquatableT_01()
+        {
+            var source =
+@"record A<T>;
+record B : A<int>;
+class Program
+{
+    static void F<T>(System.IEquatable<T> t)
+    {
+    }
+    static void M<T>()
+    {
+        F(new A<T>());
+        F(new B());
+        F<A<int>>(new B());
+        F<B>(new B());
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (11,9): error CS0411: The type arguments for method 'Program.F<T>(IEquatable<T>)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
+                //         F(new B());
+                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "F").WithArguments("Program.F<T>(System.IEquatable<T>)").WithLocation(11, 9));
+        }
+
+        [Fact]
+        public void IEquatableT_02()
+        {
+            var source =
+@"using System;
+record A;
+record B<T> : A;
+record C : B<int>;
+class Program
+{
+    static string F<T>(IEquatable<T> t)
+    {
+        return typeof(T).Name;
+    }
+    static void Main()
+    {
+        Console.WriteLine(F(new A()));
+        Console.WriteLine(F<A>(new C()));
+        Console.WriteLine(F<B<int>>(new C()));
+        Console.WriteLine(F<C>(new C()));
+    }
+}";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput:
+@"A
+A
+B`1
+C");
+        }
+
+        [Fact]
+        public void IEquatableT_03()
+        {
+            var source =
+@"using System;
+record A<T> : IEquatable<A<T>>
+{
+}
+record B : A<object>, IEquatable<A<object>>, IEquatable<B>;
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            AssertEx.Equal(new[] { "System.IEquatable<A<T>>" }, comp.GetMember<NamedTypeSymbol>("A").AllInterfacesNoUseSiteDiagnostics.ToTestDisplayStrings());
+            AssertEx.Equal(new[] { "System.IEquatable<A<System.Object>>", "System.IEquatable<B>" }, comp.GetMember<NamedTypeSymbol>("B").AllInterfacesNoUseSiteDiagnostics.ToTestDisplayStrings());
+        }
+
+        [Fact]
+        public void IEquatableT_04()
+        {
+            var source =
+@"using System;
+record A<T> : IEquatable<A<T>>
+{
+    public virtual bool Equals(A<T> other) => false;
+}
+record B : A<object>, IEquatable<A<object>>, IEquatable<B>
+{
+    public override bool Equals(A<object> other) => false;
+    public virtual bool Equals(B other) => false;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
```

Comment:
>comp.VerifyDiagnostics(); [](start = 12, length = 25)

Please run the code and observe expected methods called. Otherwise, how do we know that right methods are going to be hooked up?